src/app/hooks/useStatusForAsyncCall.ts

```diff
@@ -0,0 +1,14 @@
+import { useState } from 'react';
+
+export default <T>(call: () => Promise<T>): [boolean, () => Promise<T>] => {
+  const [status, setStatus] = useState(false);
+
+  const wrappedCall = async () => {
+    setStatus(true);
+
+    const callResult = await call();
+    setStatus(false);
```

Comment:
Instead use true or false.
Maybe we can use constants, boolean is a little bit confused.

Initial: `IDLE` or `INIT`
after wrappedCall started: `IN_PROGRESS` or `BUSY`
after async call resolved: `COMPLETE`

Reply:
OR

We still use boolean, but change `status` to `busy`.
init: `busy = fasle`,
in progress: `busy = true`,
completed: `busy = false`

Reply:
Using constant enum.